test/e2e/remediation_test.go

```diff
@@ -0,0 +1,477 @@
+	Byf("Ensuring Nodes %v consistently have ready=%s status", names, status)
+	Consistently(
+		func() error {
+			for _, node := range names {
+				if err := assertNodeStatus(ctx, c, client.ObjectKey{Namespace: namespace, Name: node}, status); err != nil {
+					return err
+				}
+			}
+			return nil
+		}, e2eConfig.GetIntervals(specName, "monitor-vm-state")...).Should(Succeed())
+}
+
+func assertNodeStatus(ctx context.Context, client client.Client, name client.ObjectKey, status v1.ConditionStatus) error {
+	node := &v1.Node{}
+	if err := client.Get(ctx, name, node); err != nil {
+		return err
+	}
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == v1.NodeReady {
+			if status == condition.Status {
+				return nil
+			} else {
+				return fmt.Errorf("Node %s has status '%s', should have '%s'", name.Name, condition.Status, status)
+			}
+		}
+	}
+	return fmt.Errorf("Node %s missing condition \"Ready\"", name.Name)
+}
+
+func waitForNodeStatus(ctx context.Context, client client.Client, name client.ObjectKey, status v1.ConditionStatus, specName string) {
+	Byf("Waiting for Node '%s' to have ready=%s status", name, status)
+	Eventually(
+		func() error { return assertNodeStatus(ctx, client, name, status) },
+		e2eConfig.GetIntervals(specName, "wait-vm-state")...,
+	).Should(Succeed())
+
+}
+
+// powerCycle tests the poweroff annotation be turning given machines off and on
+func powerCycle(ctx context.Context, c client.Client, workloadClient client.Client, machines bmhToMachineSlice, specName string) {
+	Byf("Power cycling %d machines", len(machines))
+
+	Logf("Marking %d BMHs for power off", len(machines))
+	for _, set := range machines {
+		annotateBmh(ctx, c, *set.baremetalhost, poweroffAnnotation, pointer.String(""))
+	}
+	waitForVmsState(machines.getVmNames(), shutoff, specName)
+
+	// power on
+	Logf("Marking %d BMHs for power on", len(machines))
+	for _, set := range machines {
+		annotateBmh(ctx, c, *set.baremetalhost, poweroffAnnotation, nil)
+	}
+
+	waitForVmsState(machines.getVmNames(), running, specName)
+	for _, nodeName := range machines.getNodeNames() {
+		waitForNodeStatus(ctx, workloadClient, client.ObjectKey{Namespace: "default", Name: nodeName}, v1.ConditionTrue, specName)
+	}
+
+	By("waiting for nodes to consistently have Ready status")
+	Eventually(func(g Gomega) {
+		monitorNodesStatus(g, ctx, workloadClient, "default", machines.getNodeNames(), v1.ConditionTrue, specName)
+	}, e2eConfig.GetIntervals(specName, "wait-machine-remediation")...).Should(Succeed())
+}
+
+// cleanObjectMeta clears object meta after copying from the original object
+func cleanObjectMeta(om *metav1.ObjectMeta) {
+	om.UID = ""
+	om.Finalizers = nil
+	om.ManagedFields = nil
+	om.ResourceVersion = ""
+	om.OwnerReferences = nil
+}
+
+func getMachine(ctx context.Context, c client.Client, name client.ObjectKey) (result clusterv1.Machine) {
```

Comment:
I will add a description on this function since its name doesn't fully convey its functionality, but I think it is not necessary to add to other functions because all of them are private function, and the function name describe pretty good.